pkg/tools/security/token_test.go

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security_test
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/asn1"
+	"encoding/pem"
+	"math/big"
+	"testing"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+	"github.com/stretchr/testify/require"
+
+	"github.com/networkservicemesh/sdk/pkg/tools/security"
+)
+
+const (
+	spiffeID    = "spiffe://test.com/workload"
+	nameTypeURI = 6
+)
+
+var (
+	testCA             tls.Certificate
+	testTLSCertificate tls.Certificate
+)
+
+func init() {
+	var err error
+	testCA, err = generateCA()
+	if err != nil {
+		panic(err)
+	}
+
+	testTLSCertificate, err = generateKeyPair(spiffeID, "test.com", &testCA)
+	if err != nil {
+		panic(err)
+	}
+}
```

Comment:
Is it possible to use testify/suite for setup tests? Tesitfy suites can have setup/teardown methods. take a look at https://github.com/stretchr/testify/blob/master/suite/interfaces.go#L20

Reply:
Fixed